pkg/operator/operator.go

```diff
@@ -35,23 +35,28 @@ type Operator struct {
 }
 
 // New creates (but does not start) a new operator from configuration.
-func New(config operatorconfig.Config) (*Operator, error) {
-	kubeConfig, err := kconfig.GetConfig()
-	if err != nil {
-		return nil, fmt.Errorf("failed to get kube config: %v", err)
-	}
-	scheme := operatorclient.GetScheme()
+func New(config operatorconfig.Config, kubeConfig *rest.Config) (*Operator, error) {
 	operatorManager, err := manager.New(kubeConfig, manager.Options{
-		Scheme:    scheme,
-		Namespace: "openshift-dns",
+		Scheme:         operatorclient.GetScheme(),
+		Namespace:      "openshift-dns",
+		MapperProvider: operatorutil.NewDynamicRESTMapper,
+		// Use a non-caching client everywhere. The default split client does not
+		// promise to invalidate the cache during writes (nor does it promise
+		// sequential create/get coherence), and we have code which (probably
+		// incorrectly) assumes a get immediately following a create/update will
+		// return the updated resource. All client consumers will need audited to
+		// ensure they are tolerant of stale data (or we need a cache or client that
+		// makes stronger coherence guarantees).
+		NewClient: func(_ cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
```

Comment:
Is the operator caches field no longer needed? https://github.com/openshift/cluster-dns-operator/blob/master/pkg/operator/operator.go#L33

Reply:
Nothing's using the cache yet, but soon will, so "reserved for future use" — if you'd rather I delete it until needed let me know

Reply:
Thanks for the explanation. I think it's fine to stay. Just trying to make sure I understand what's going on.